demos/random_cut_demo/src/main/scala/io/hydrosphere/rrcf/impl/ForestParams.scala

```diff
@@ -9,17 +9,17 @@ import scala.io.Source
 import scala.util.control.NonFatal
 
 case class ForestParams(
-  treesNum: Int = 5,
-  samplesNum: Int = 100,
-  shingleSize: Int = 4,
-  timeDecay: Long = 10000,
+  treesNum: Int = 300,
+  samplesNum: Int = 300,
+  shingleSize: Int = 2,
+  timeDecay: Long = 1000,
   warmupScore: Double = -1
 ) {
   override def toString: String = {
     s"Number of trees: $treesNum; Samples per tree: $samplesNum; Shingle size: $shingleSize; Time-Decay: $timeDecay"
   }
 
-  def warmupPointsNum = samplesNum * treesNum + shingleSize
+  def warmupPointsNum = samplesNum + shingleSize - 1

```

Comment:
what's the reason behind this?

Reply:
@KineticCookie  -1?

Reply:
num_windows = points - window_size + 1
Old formula was generating 1 sliding window (or shingle in other words) more than needed. This was not raising exceptions since this points was just ommited.